framework/src/mesh/MooseMesh.C

```diff
@@ -2570,10 +2570,16 @@ MooseMesh::effectiveSpatialDimension() const
 unsigned int
 MooseMesh::getBlocksMaxDimension(const std::vector<SubdomainName> & blocks) const
 {
+  const auto & mesh = getMesh();
+
+  // Take a shortcut if possible
+  if (const auto & elem_dims = mesh.elem_dimensions(); mesh.is_prepared() && elem_dims.size() == 1)
```

Comment:
Even if `elem_dims.size() > 1`, can't we still do an early `return *elem_dims.rbegin()`?

Reply:
Our cache data is for the entire mesh. This is filtering for given blocks